### PR TITLE
[External solvers] Remove old FEAST version

### PR DESCRIPTION
**Pull request by qaumann**
As requested in #6482 this removes the old version of FEAST. The tests in `StructuralMechancisApplication` are migrated to the new FEAST as of #6721. The old interface is apparently also used in `DamApplication` and `SolidMechanicsApplication`. Should this also be adapted or is it not used actively there? Who would be in charge?

@loumalouomega I moved the `feast_condition_number_uitility.h` to `EigenSolversApplication` and adapted it to the new FEAST version. I did not find any tests though, so I am not 100% sure if I did everything correct

### CODE REVIEW

**Comment by qaumann** [comment]
@loumalouomega many thanks for adapting the utility.

As the dependency on the old FEAST has been removed from `StructuralMechanicsApplication` in #6721 and from `DamApplication` in #6726, I think this is ready for review now.

File: applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_eigensolver.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def _create_mechanical_solution_strategy(self):
         if solver_type == "eigen_eigensystem":
             mass_matrix_diagonal_value = 0.0
             stiffness_matrix_diagonal_value = 1.0
-        elif solver_type == "feast" or solver_type == "eigen_feast":
+        elif solver_type == "eigen_feast":
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_eigensolver.py]: since now the feast in the eigensolversapp is the only one I would use the name `feast`
```suggestion
        elif solver_type == "feast":
```

Comment (philbucher) [reply applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_eigensolver.py]: can you please update the remaining occurences too?

File: kratos/python_scripts/python_linear_solver_factory.py [hunk 2]
```diff
@@ -83,8 +82,7 @@ def __CheckIfSolverTypeIsDeprecated(config):
         "SuperLUSolver"          : "super_lu",
         "SuperLUIterativeSolver" : "super_lu_iterative",
         "PastixSolver"           : "pastix",
-        "PastixComplexSolver"    : "pastix_complex",
-        "FEASTSolver"            : "feast"
+        "PastixComplexSolver"    : "pastix_complex"
```
Comment (philbucher) [inline kratos/python_scripts/python_linear_solver_factory.py]: can you please revert the changes in this file?
I saw that it needs some cleanup, which I would like to do in a separate PR

File: applications/StructuralMechanicsApplication/python_scripts/automatic_rayleigh_parameters_computation_process.py [hunk 3]
```diff
@@ -159,27 +159,7 @@ def _auxiliar_eigen_settings(self, solver_type):
...
-                    "linear_solver_settings": {
-                        "solver_type": "skyline_lu_complex"
-                    }
-                }
-            }
-            """)
-            self.mass_matrix_diagonal_value = 1.0
-            self.stiffness_matrix_diagonal_value = -1.0
-        elif solver_type == "eigen_feast":
+        if solver_type == "eigen_feast":
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/python_scripts/automatic_rayleigh_parameters_computation_process.py]:
```suggestion
        if solver_type == "feast":
```

**Review by philbucher — COMMENTED** [review]
minor comments but otherwise very cool!
Didn't expect this to be so smoothly

File: kratos/python_scripts/python_linear_solver_factory.py (hunk 2 above)
Comment (philbucher) [reply kratos/python_scripts/python_linear_solver_factory.py]: can you please do this, then I think it is ready

Comment (qaumann) [reply kratos/python_scripts/python_linear_solver_factory.py]: sorry, missed that

**Review by philbucher — APPROVED** [review]
awesome, thanks!

FYI @KratosMultiphysics/technical-committee

**Comment by qaumann** [comment]
@philbucher I saw there are some examples in the examples repo using the old FEAST interface. Should they be changed before merging?

**Comment by philbucher** [comment]
> @philbucher I saw there are some examples in the examples repo using the old FEAST interface. Should they be changed before merging?

I would merge this now, then you can update the examples (thx!)
=> you don't need to make PRs in the examples repo
